lib/galaxy/workflow/modules.py

```diff
@@ -598,6 +598,9 @@ def _find_collections_to_match(self, progress: "WorkflowProgress", step, all_inp
                     effective_input_collection_type,
                     dataset_collection_type_descriptions,
                 )
+                if not progress.subworkflow_structure and history_query.direct_match(data):
+                    continue
```

Comment:
```suggestion
                    # TODO: test paired_or_single map over of subworkflow in invocation.
                    continue
```
Or are you convinced this is correct now ?

Reply:
I'm convinced you're much smarter than me and I cannot break it 😅. I'll make a note to add some test cases in this direction but I don't think we need the TODO simply because I'm not capable of keeping this in my head.